File: DSA/Cpp/test/dsa/lib/math/ds/median_test.cpp
#include <dsa/lib/math/ds/median.hpp>

namespace dsa::lib::math::ds {
namespace {

double findMedianSortedArray(std::vector<int> &&nums1, std::vector<int> &&nums2) {
    return Median().findMedianSortedArrays(nums1, nums2);
}
TEST(FindMedianSortedArraysTest, case1) {
    EXPECT_FLOAT_EQ(2.0, findMedianSortedArray({1, 3}, {2}));
    EXPECT_FLOAT_EQ(2.5, findMedianSortedArray({1, 2}, {3, 4}));
}

}
}

File: DSA/Java/test/dsa/lib/math/ds/MedianTest.java
package dsa.lib.math.ds;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class MedianTest {
    public static class FindMedianSortedArrayTest {
        @Test
        public void case1() {
            assertEquals(2.0, findMedianSortedArray(new int[]{1, 3}, new int[]{2}), .1);
            assertEquals(2.5, findMedianSortedArray(new int[]{1, 2}, new int[]{3, 4}), .1);
        }

        private double findMedianSortedArray(int[] nums1, int[] nums2) {
            return new Median().findMedianSortedArrays(nums1, nums2);
        }
    }
}
